### DISCUSSION TITLE
Weblate and AWS SES

### DISCUSSION THREAD

**Discussion by AKIvan** [WeblateOrg/weblate#10287] [Q&A]
Hi all, 

I have configured the SES and i have test it with boto3 - seams to work. I have defined the env. variables for the weblate to use it the Secret key and access key provided by the ses when configured, and from the logs im seeing that everything looks ok (or im missing something)

`

October 27, 2023 at 12:53 (UTC+2:00) | celery-celery stderr \| [2023-10-27 10:53:38,050: INFO/MainProcess] Task weblate.utils.tasks.heartbeat[6c2a16a2-0e4f-4a5d-9f87-bdd86893d535] received | Weblate
-- | -- | --
October 27, 2023 at 12:53 (UTC+2:00) | nginx stdout \| 10.0.8.57 - - [27/Oct/2023:10:53:29 +0000] "GET / HTTP/1.1" 200 6294 "-" "ELB-HealthChecker/2.0" | Weblate
October 27, 2023 at 12:53 (UTC+2:00) | nginx stdout \| 10.0.105.137 - - [27/Oct/2023:10:53:08 +0000] "GET / HTTP/1.1" 200 6293 "-" "ELB-HealthChecker/2.0" | Weblate
October 27, 2023 at 12:52 (UTC+2:00) | nginx stdout \| 10.0.8.57 - - [27/Oct/2023:10:52:59 +0000] "GET / HTTP/1.1" 200 6296 "-" "ELB-HealthChecker/2.0" | Weblate
October 27, 2023 at 12:52 (UTC+2:00) | celery-notify stderr \| [2023-10-27 10:52:40,233: INFO/MainProcess] Task weblate.accounts.tasks.send_mails[e8f755b2-b240-454c-a9a0-155a39991df8] received | Weblate
October 27, 2023 at 12:52 (UTC+2:00) | celery-notify stderr \| [2023-10-27 10:52:40,229: INFO/MainProcess] Task weblate.accounts.tasks.send_mails[0efe9be9-426a-4399-a957-9e63fc36f83d] received | Weblate

`
I still have not receive anything. Also what is the variable or email that is used as verified e-mail from SES side ? Is it the weblate admin e-mail or ? Or is there a different way of configuring ses?

**Comment by nijel** [comment]
There is no SES integration in Weblate yes, needed steps were already mentioned in https://github.com/WeblateOrg/weblate/discussions/9843#discussioncomment-6933747

**Reply by AKIvan** [reply]
I thought installing the library and configure the environments will enable Weblate to send e-mails and stuff using SES ? Did i misunderstand you ?

**Reply by AKIvan** [reply]
And yes i have installed the library and create a new container of it. From the logs seams like its not processed do i missing something ?

**Reply by nijel** [reply]
The library needs to be configured in Python settings, you can do it in setttings-override.py

**Reply by AKIvan** [reply]
setttings-override.py ? settings.py from weblate or django-ses ?

**Reply by AKIvan** [reply]
Even if i configure if Weblate does not have integrated this process it wont work. I need weblate to send e-mails to the users to use all the functionality when translations are done etc. So SES seams to be out of option as its not fully integrated or im wrong? Any other way of configuring weblate to send e-mails ?

**Reply by nijel** [reply]
Installing the package and configuring as described in https://github.com/django-ses/django-ses#getting-going should work just fine.

Weblate Docker image supports only SMTP out of the box. You can customize it to use any Django backend manually, see https://docs.weblate.org/en/latest/admin/install/docker.html#configuration-beyond-environment-variables

**Reply by nijel** [reply]
https://github.com/WeblateOrg/weblate/issues/10341 describes what is needed for native integration in Weblate Docker image.

**Reply by AKIvan** [reply]
So this is my dockerfile 

```FROM weblate/weblate:5.0.2.2

COPY requirements.txt /tmp/
RUN pip install --no-cache-dir -r /tmp/requirements.txt
```
the requirements.txt have
`django-ses==3.4.1`


On the ECS Side i have defined all possible env variables with the secrets and endpoint and access key. (take in consideration using boto3 i have test this secrets and access key and they worked). 
The following env variable defined:

```WEBLATE_EMAIL_HOST
WEBLATE_EMAIL_HOST_USER
WEBLATE_EMAIL_PASSWORD
WEBLATE_EMAIL_HOST_PASSWORD

AWS_ACCESS_KEY_ID
AWS_SES_REGION_ENDPOINT
AWS_SECRET_ACCESS_KEY
```

After Deployment i try to test it by adding different mail to my profile. From profile > settings > Account and "Add new e-mail address". I should receive some e-mail using my private account but nothing. 
Am im missing something here ?

**Reply by nijel** [reply]
Again: There is nothing parsing your AWS_SES environment variables. It needs to be implemented as described in https://github.com/WeblateOrg/weblate/issues/10341. Or use https://docs.weblate.org/en/latest/admin/install/docker.html#overriding-settings-from-the-data-volume
